other/huffmancoding/huffmancoding.go

```diff
@@ -0,0 +1,123 @@
+// huffman.go
+// description: Implements Huffman compression, encoding and decoding
+// details:
+// We implement the linear-time 2-queue method described here https://en.wikipedia.org/wiki/Huffman_coding.
+// It assumes that the list of symbol-frequencies is sorted.
+// author(s) [pedromsrocha](https://github.com/pedromsrocha)
+// see also huffmancoding_test.go
+
+// Package huffman provides primitives for doing Huffman compression, encodding and decoding.
+package huffman
+
+// A Node of an Huffman tree, which can either be a leaf or an internal node.
+// Each node has a weight.
+// A leaf node has an associated symbol, but no children (i.e., left == right == nil).
+// A parent node has a left and right child and no symbol (i.e., symbol == -1).
+type Node struct {
+	left   *Node
+	right  *Node
+	symbol rune
+	weight int
+}
+
+// A SymbolFreq is a pair of a symbol and its associated frequency.
+type SymbolFreq struct {
+	symbol rune
+	freq   int
+}
+
+// The algorithm operates on []SymbolFreq sorted by frequency
+type ByFreq []SymbolFreq
+
+func (x ByFreq) Len() int           { return len(x) }
+func (x ByFreq) Less(i, j int) bool { return x[i].freq < x[j].freq }
+func (x ByFreq) Swap(i, j int)      { x[i], x[j] = x[j], x[i] }
```

Comment:
From what I can tell, you're making an assumption that the provided list of symbol-frequency is already sorted, therefore I don't see why you this type has to satisfy the `sort.Interface`. And even more so when you're not even importing the `sort` package....

Reply:
You're completely right! Commit 318d135 moves the sort interface method instantiation to the test file, the place where it's used.